src/omlt/linear_tree/lt_definition.py

```diff
@@ -0,0 +1,408 @@
+            Engineering
+        * Misic, V. "Optimization of tree ensembles."
+             Operations Research 68.5 (2020): 1605-1624.
+        * Mistry, M., et al. "Mixed-integer convex nonlinear optimization
+             with gradient-boosted trees embedded." INFORMS Journal on
+             Computing (2020).
+    """
+
+    def __init__(
+        self,
+        lt_regressor,
+        scaling_object=None,
+        scaled_input_bounds=None,
+        unscaled_input_bounds=None,
+    ):
+        """Create a LinearTreeDefinition object and define attributes based on the
+        trained linear model decision tree.
+
+        Arguments:
+            lt_regressor -- A LinearTreeRegressor model that is trained by the
+                linear-tree package
+
+        Keyword Arguments:
+            scaling_object --A scaling object to specify the scaling parameters
+                for the linear model tree inputs and outputs. If None, then no
+                scaling is performed. (default: {None})
+            scaled_input_bounds -- A dict that contains the bounds on the scaled
+                variables (the direct inputs to the tree). If None, then the
+                user must specify the bounds via the input_bounds argument.
+                (default: {None})
+            unscaled_input_bounds -- A dict that contains the bounds on the
+                variables (the direct inputs to the tree). If None, then the
+                user must specify the scaled bounds via the scaled_input_bounds
+                argument. (default: {None})
+
+        Raises:
+            Exception: Input bounds required. If unscaled_input_bounds and
+                scaled_input_bounds is None, raise Exception.
+        """
+        self.__model = lt_regressor
+        self.__scaling_object = scaling_object
+
+        # Process input bounds to insure scaled input bounds exist for formulations
+        if scaled_input_bounds is None:
+            if unscaled_input_bounds is not None and scaling_object is not None:
+                lbs = scaling_object.get_scaled_input_expressions(
+                    {k: t[0] for k, t in unscaled_input_bounds.items()}
+                )
+                ubs = scaling_object.get_scaled_input_expressions(
+                    {k: t[1] for k, t in unscaled_input_bounds.items()}
+                )
+
+                scaled_input_bounds = {
+                    k: (lbs[k], ubs[k]) for k in unscaled_input_bounds.keys()
+                }
+
+            # If unscaled input bounds provided and no scaler provided, scaled
+            # input bounds = unscaled input bounds
+            elif unscaled_input_bounds is not None and scaling_object is None:
+                scaled_input_bounds = unscaled_input_bounds
+            elif unscaled_input_bounds is None:
+                raise ValueError(
+                    "Input Bounds needed to represent linear trees as MIPs"
+                )
+
```

Comment:
It is unfortunate that we need to copy this from network_definition. A sign that we don't yet have the best abstraction. Let's consider this during a future refactoring.

Reply:
Agreed